test/e2e/set/scyllacluster/scyllacluster_pv.go

```diff
@@ -5,35 +5,272 @@ package scyllacluster
+								PersistentVolumeReclaimPolicy: pv.Spec.PersistentVolumeReclaimPolicy,
+								MountOptions:                  pv.Spec.MountOptions,
+								VolumeMode:                    pv.Spec.VolumeMode,
+								StorageClassName:              testStorageClassName,
+								NodeAffinity:                  nil,
+							},
+							Status: corev1.PersistentVolumeStatus{
+								Phase: pv.Status.Phase,
+							},
+						}
+
+						if pvClone.Spec.PersistentVolumeSource.Local != nil {
+							pvClone.Spec.PersistentVolumeSource.HostPath = &corev1.HostPathVolumeSource{
+								Path: pvClone.Spec.PersistentVolumeSource.Local.Path,
+							}
+							pvClone.Spec.PersistentVolumeSource.Local = nil
+						}
+						framework.Infof("Creating clone PV for %q", pv.Name)
+						_, err = f.KubeAdminClient().CoreV1().PersistentVolumes().Create(ctx, pvClone, metav1.CreateOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						realPvcCopy := realPvc.DeepCopy()
+						realPvcCopy.Spec.VolumeName = pvClone.Name
+
+						patch, err := helpers.CreateTwoWayMergePatch(realPvc, realPvcCopy)
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						framework.Infof("Binding ScyllaCluster PVC %q with clone PV %q", realPvc.Name, pvClone.Name)
+						_, err = f.KubeClient().CoreV1().PersistentVolumeClaims(realPvc.Namespace).Patch(ctx, realPvc.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+					}
+				case watch.Deleted:
+					// If original PVC is deleted, delete the clone too.
+					pvc := e.Object.(*corev1.PersistentVolumeClaim)
+					if !strings.HasPrefix(pvc.Name, "clone-") {
+						clonePVCName := fmt.Sprintf("clone-%s", pvc.Name)
+
+						framework.Infof("Deleting clone PVC %q", clonePVCName)
+						clonePVC, err := f.KubeClient().CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(ctx, clonePVCName, metav1.GetOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						err = f.KubeClient().CoreV1().PersistentVolumeClaims(clonePVC.Namespace).Delete(ctx, clonePVC.Name, metav1.DeleteOptions{
+							Preconditions: metav1.NewUIDPreconditions(string(clonePVC.UID)),
+						})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						consumerPodName := fmt.Sprintf("consumer-%s", clonePVC.Name)
+						framework.Infof("Deleting Pod consumer of clone PVC %q", consumerPodName)
+
+						consumerPod, err := f.KubeClient().CoreV1().Pods(clonePVC.Namespace).Get(ctx, consumerPodName, metav1.GetOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						err = f.KubeClient().CoreV1().Pods(consumerPod.Namespace).Delete(ctx, consumerPod.Name, metav1.DeleteOptions{
+							Preconditions: metav1.NewUIDPreconditions(string(consumerPod.UID)),
+						})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						err = framework.WaitForObjectDeletion(ctx, f.DynamicAdminClient(), corev1.SchemeGroupVersion.WithResource("persistentvolumeclaims"), f.Namespace(), clonePVC.Name, &clonePVC.UID)
```

Comment:
why is waiting here important here?it namespaced and we'd wait later on anyways, right?

Reply:
To not handle AlreadyExists err in clone PVC creation after node replace

Reply:
could use use UUID instead of a name to avoid that wait?

Reply:
I'm using that name in L181 to avoid caching the mapping.